### PR TITLE
Semantic checks for OpenMP teams directive

### PR DESCRIPTION
**Pull request by DavidTruby**
Simple semantic checks for omp teams

### CODE REVIEW

File: lib/semantics/check-omp-structure.cc [hunk 1]
```diff
@@ -404,6 +404,15 @@ void OmpStructureChecker::Enter(const parser::OpenMPBlockConstruct &x) {
         OmpClause::DEVICE, OmpClause::DEFAULTMAP, OmpClause::NOWAIT};
     SetContextAllowedOnce(allowedOnce);
   } break;
+  case parser::OmpBlockDirective::Directive::Teams: {
```
Comment (kiranchandramohan) [inline lib/semantics/check-omp-structure.cc]: For other directives, the clauses are listed in comments before the case statement. Could you do the same here?

Comment (DavidTruby) [reply lib/semantics/check-omp-structure.cc]: Good point. I even made a mental note to do this when I went off to write the tests to check it, then never came back and did it..

**Comment by ichoyjx** [comment]
Do you think whether the following restriction can be done in this PR?
> If specified, a teams construct must be contained within a target construct. That target construct must contain no statements, declarations or directives outside of the teams construct.

**Comment by DavidTruby** [comment]
I don't think this is verifiable in the general case? As in, I believe it's permissible to have a target construct containing a function call and a teams construct inside that function. In which case the teams construct can't actually check that it appears inside a target region.

Possibly we could add some metadata to a function signature in the module file that signifies that a function containing a teams directive can only be called in a target region?

**Comment by DavidTruby** [comment]
I've just realised that I misread the restriction, and the target construct cannot contain other statements (and therefore can't contain a function call as in the example I used before). So yes, this check should be doable, I'll add it

**Comment by kiranchandramohan** [comment]
There is a card for nesting of regions (2.17) in the projects page. It gave the impression that this is a separate block of work.

**Comment by DavidTruby** [comment]
I agree that this should go in a separate PR once all the directives are done, it requires a little more machinery that will be common across any nesting checks so to me it makes sense to do that all at once

**Pull request merged**